packages/babel-plugin-transform-react-jsx/src/create-plugin.ts

```diff
@@ -39,6 +39,15 @@ const get = (pass: PluginPass, name: string) =>
 const set = (pass: PluginPass, name: string, v: any) =>
   pass.set(`@babel/plugin-react-jsx/${name}`, v);
 
+function hasProto(node: t.ObjectExpression) {
+  return node.properties.some(
+    value =>
+      t.isObjectProperty(value, { computed: false }) &&
```

Comment:
```suggestion
      t.isObjectProperty(value, { computed: false, shorthand: false }) &&
```

Shorthand `__proto__` doesn't set the prototype.